### ISSUE TITLE
Custom NPCs Issue

### ISSUE THREAD

**Issue by jordanasch** [ForgeEssentials/ForgeEssentials#2293]
Hey there guys, I have a problem where I cannot issue commands through custom npcs dialog actions I use the 1.10.2 version.

I have tried lots of perm functions nothing I have tried works.

This is how I can recreate the problem and/or need it to function:
1. Create a forge server I use forge-1.10.2-12.18.3.2281
2. Add FE Server and Custom NPCs (CustomNPCs_1.10.2(25may17)) to server
3. When in the server, create an NPC with the NPC Wand tool
4. When in that same NPC interface go to the last tab on the right to create some dialogs. When in the dialog creation interface add a new category and enter it. Enter what ever you want for the name, and where it says "dialog options" click that and click the first empty space. Then where it says "Dialog" click that until it says "Command". Then enter in any command you like that is easy to guage if it was successful(I use tp)
5. After you are done creating the dialog, you must go to the second last tab on the right of the custom npcs creation interface and bind the dialog to the NPC you created (Make sure you are still in the same interface you have been in the whole time or just make sure that you right click the same NPC you want to have the dialog with the NPC wand)
6. Test the dialog by right clicking on the NPC with your fist.

**Comment by mattparksjr** [comment]
Try fe.tp or fe.teleport

**Comment by Winsworth** [comment]
Any more luck with this issue? It feels like it's a permission issue in so much as custom npcs mod npcs aren't able to use commands. Whether thats because they don't have a permission group i don't know. I do know that the commands work without issue if Forge essentials is removed which is why i feel it's a problem assigning permissions to the custom npcs. 

I'm also running 1.10.2 version and have been pulling my hair out trying to get it to work all afternoon. 
The custom NPC logs are the only logs i can find any trace of the issue in. : 

```
[23:30:15][SEVERE/CustomNPCs][java.util.concurrent.Executors$RunnableAdapter:511] java.lang.ClassCastException: noppes.npcs.NoppesUtilServer$1 cannot be cast to net.minecraft.entity.player.EntityPlayer
	at net.minecraft.command.CommandHandler.redirect$hasPermission$zzb000(CommandHandler.java:597)
	at net.minecraft.command.CommandHandler.func_71556_a(CommandHandler.java:48)
	at noppes.npcs.NoppesUtilServer.runCommand(NoppesUtilServer.java:232)
	at noppes.npcs.NoppesUtilServer.runCommand(NoppesUtilServer.java:161)
	at noppes.npcs.NoppesUtilPlayer.dialogSelected(NoppesUtilPlayer.java:292)
	at noppes.npcs.PacketHandlerPlayer.player(PacketHandlerPlayer.java:161)
	at noppes.npcs.PacketHandlerPlayer.access$000(PacketHandlerPlayer.java:41)
	at noppes.npcs.PacketHandlerPlayer$1.run(PacketHandlerPlayer.java:53)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at net.minecraft.util.Util.func_181617_a(SourceFile:45)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:668)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
	at java.lang.Thread.run(Thread.java:748)
```
As i say i know the commands work without FE. So it feels like its something in the way fe handles the permissions.

Any help would be greatly appreciated, 

Thanks in advance!

**Comment by Zethalion** [comment]
As we moved to 1.12.2, is this still an issue? Closing this issue in 7 days if no response @Aschy2

**Comment by Parritoso** [comment]
I know it's not my issue, but the problem is still in 1.12.2

**Comment by vin0m** [comment]
Yes this is still an issue in 1.12.2.  Is there any way to have FE ignore Custom NPC entities/scripted blocks, etc? It appears I need to grant permissions for every NPC/scripted block. That would be extremely difficult.

**Comment by Zethalion** [comment]
Not to be of any annoyance but this issue should have been closed last year. As there is still an issue regarding this one, there will be needed to open a new issue based on the current FE 1.12.2 build 56. Reference to this issue might be ignored as 1.10 and 1.12.2 are too different.

**Comment by Tennik** [comment]
Try to remove a / before the tp and use @dp instead of @p. Like "tp @dp 2000 100 1000". 
That resolved the same issue with 1.12.2 version of mod on my server
